### PR TITLE
RC 0.0.47

### PR DESCRIPTION
**Pull request by carletex**
Since we have a few things to merge, we thought it'd be nice to test a couple of things:

1. Use of release candidate branch (that we will merge to main) and we can test all the changes together
2. Publishing a `beta` package in npm.

This PR includes #2 #3 #14 #45 #47 #50 #51

### CODE REVIEW

File: .changeset/eleven-ways-count.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+"create-eth": patch
+---
+
+change listr to listr2
```
Comment (rin-st) [inline .changeset/eleven-ways-count.md:5]: copies this change https://github.com/scaffold-eth/create-eth/pull/53/files#diff-32998dd93341cf2d3e267eb301f7e3b4d41155a54ec890935f4bc161b6700461R6

Comment (rin-st) [reply .changeset/eleven-ways-count.md:5]: I think this file can be removed

**Comment by carletex** [comment]
Beta version published on:

`npx create-eth@0.0.47-beta.1`

More info about pre-releases here: https://github.com/changesets/changesets/blob/main/docs/prereleases.md (we should all of this stuff on #6) . Also add the more verbose option to use npx: `npm create eth@<version>` (the space between `create` and `eth` should be there, not a typo.)

File: .changeset/eleven-ways-count.md (hunk 1 above)
Comment (carletex) [reply .changeset/eleven-ways-count.md]: Done! https://github.com/scaffold-eth/create-eth/pull/53/commits/ac1dbd9822ff59298ba86e88b0affca18d307319

**Comment by carletex** [comment]
Thinking about merging vs squash and merging here (with this release branch model).

I think if squash and merge, the existing PRs won't close (but they history will be clean). If we just merge, the PRs will close... but the git history :sweat: 

But then...I'm thinking that if we are using this system, we should be merging always against a release branch, not main.... so we shouldn't have this problem, right?

So maybe test this, squash & merge... and close the PRs manually?

**Comment by rin-st** [comment]
>  If we just merge, the PRs will close... 

Not sure I understand how it works. Why they will close? Because all the commits will already be on main? If yes, everything else seems reasonable. Only one question

> But then...I'm thinking that if we are using this system, we should be merging always against a release branch, not main.... so we shouldn't have this problem, right?

We don't know at the start if it will be patch, minor or major. So maybe use `dev` branch? Merge to `dev` and then to`rc` branches from `dev` when ready?

btw, A new commit added to #45 . Not sure it's completed though

**Comment by carletex** [comment]
> Why they will close? Because all the commits will already be on main?

Yep!

> We don't know at the start if it will be patch, minor or major. So maybe use dev branch? Merge to dev and then torc branches from dev when ready? After it squash rc to main.

Yeah, not sure. Let's keep exploring as we go and decide. Thanks!

> btw, A new commit added to https://github.com/scaffold-eth/create-eth/pull/45 . Not sure it's completed though

Renaming the file? Yep, that was added here.

---

Also added the eip712 curated extension in this PR.

**Comment by rin-st** [comment]
> Renaming the file? Yep, that was added here.

hmm, I don't see that commit and see this in files tree

image

> Also added the eip712 curated extension in this PR.

nit: maybe name it `eip-712` instead of `eip712`? So it will be consistent with branch name on extensions and with naming [here](https://github.com/ethereum/EIPs/tree/master/EIPS)?

---
tested #50, #51, -e `eip712` and also #45 and everything works well for me

**Review by rin-st — APPROVED** [review]

**Comment by carletex** [comment]
> hmm, I don't see that commit and see this in files tree

Oh sorry, created this after Shiv pushed, but forgot to pull it first :D Thanks for adding it Shiv!

> nit: maybe name it eip-712 instead of eip712? So it will be consistent with branch name on extensions and with naming [here](https://github.com/ethereum/EIPs/tree/master/EIPS)?

Yeah, this makes sense.

**Review by technophile-04 — APPROVED** [review]
Tested nicely and everything works nicely !! also all the extensions, also tried out : 

```shell
 npx create-eth@0.0.47-beta.1 -e technophile-04/eip-5792:foundry
```

works nicely ! Thanks all !!

**Pull request merged**